src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/person/PersonService.kt

```diff
@@ -34,65 +30,40 @@ class PersonService(
   private val personRepository: PersonRepository,
   private val readWriteLockService: ReadWriteLockService,
   private val searchService: SearchService,
-  private val matchService: MatchService,
   private val personKeyService: PersonKeyService,
   private val queueService: QueueService,
   @Value("\${retry.delay}") private val retryDelay: Long,
 ) {
 
-  fun processMessage(person: Person, event: String? = null, linkRecord: Boolean = true, callback: (isAboveSelfMatchThreshold: Boolean) -> PersonEntity?): PersonEntity = runBlocking {
+  fun processMessage(person: Person, event: String? = null, linkRecord: Boolean = true, callback: () -> PersonEntity?): PersonEntity = runBlocking {
     runWithRetry(MAX_ATTEMPTS, retryDelay, ENTITY_RETRY_EXCEPTIONS) {
       readWriteLockService.withWriteLock(person.sourceSystemType) { return@withWriteLock processPerson(person, event, linkRecord, callback) }
     }
   }
 
-  private fun processPerson(person: Person, event: String?, linkRecord: Boolean, callback: (isAboveSelfMatchThreshold: Boolean) -> PersonEntity?): PersonEntity {
-    processSelfMatchScore(person)
-    val existingPersonEntity: PersonEntity? = callback(person.isAboveMatchScoreThreshold)
+  private fun processPerson(person: Person, event: String?, linkRecord: Boolean, callback: () -> PersonEntity?): PersonEntity {
+    val existingPersonEntity: PersonEntity? = callback()
     val personEntity: PersonEntity = when {
       (existingPersonEntity == null) -> handlePersonCreation(person, event, linkRecord)
       else -> handlePersonUpdate(person, existingPersonEntity, event)
     }
     return personEntity
   }
 
-  private fun processSelfMatchScore(person: Person) {
-    val (isAboveSelfMatchThreshold, selfMatchScore) = matchService.getSelfMatchScore(PersonSearchCriteria.from(person))
-    person.selfMatchScore = selfMatchScore
-    person.isAboveMatchScoreThreshold = isAboveSelfMatchThreshold
-    telemetryService.trackPersonEvent(
-      CPR_SELF_MATCH,
-      person,
-      mapOf(
-        EventKeys.PROBABILITY_SCORE to selfMatchScore.toString(),
-        EventKeys.IS_ABOVE_SELF_MATCH_THRESHOLD to isAboveSelfMatchThreshold.toString(),
-      ),
-    )
-  }
-
   private fun handlePersonCreation(person: Person, event: String?, linkRecord: Boolean): PersonEntity {
     if (isUpdateEvent(event)) {
       telemetryService.trackPersonEvent(CPR_UPDATE_RECORD_DOES_NOT_EXIST, person)
     }
     val personEntity = createPersonEntity(person)
     val personKey: PersonKeyEntity? = when {
-      linkRecord && person.isAboveMatchScoreThreshold -> personKeyService.getPersonKey(personEntity)
-      else -> handleLowSelfMatchScore(person)
+      linkRecord -> personKeyService.getPersonKey(personEntity)
```

Comment:
I'm going to refactor unmerge service in a separate PR so it doesn't spill over complexity / uncouple the components